libr/anal/fcn.c

```diff
@@ -1915,23 +1915,15 @@ R_API bool r_anal_fcn_add_bb(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut64 si
 		if (bb) {
 			r_list_delete_data (fcn->bbs, bb);
 		}
-		ut8 *bbuf = malloc (size);
-		if (!bbuf) {
-			eprintf ("malloc failed\n");
-			return false;
-		}
-		anal->iob.read_at (anal->iob.io, addr, bbuf, size);
-		fcn_recurse (anal, fcn, addr, bbuf, size, 1);
+		fcn_recurse (anal, fcn, addr, 1);
```

Comment:
this is one of the problems in r2r (see XX ` db/cmd/cmd_af afb+ with wrong size`). Before, the `size` was used to limit the size of the basicblock, now instead since you removed `len` in `fcn_recurse`, there is no way to know if you want a smaller bb size.

Reply:
actually i find out the recurseAt macro is not used in all situations, this is a bit confusing and inconsistent, but i understand there are some reasons behind this. i added a macro to toggle that behaviour for proper debugging. also found that some tests are wrong :<<<< chk this:

```
s 0x004017c0
af+ 0x004017c0 fcn2.0x004017c0
afb+ 0x4017c0 0x4017c0 9 0x4017dd 0x4017cd
afi~size
```

its specifying that the blocksize is 9 bytes... its hardcoding this value, but the code in ithere if we stick at 9 it will fail to disassemble, because the rgiht value here is 6

```
[0x004017c0]> pD 6
            0x004017c0      51             push ecx
            0x004017c1      89e1           mov ecx, esp
            0x004017c3      83c108         add ecx, 8
```

so maybe all the broken tests are just bad tests


btw ,aab with this branch works fine..

Reply:
No I think the tests are right. The whole point of using `afb+` manually is that you know more than r2, so you want to tell r2 what is a basic block exactly. If the user says it's 9 bytes, than 9 bytes it shall be IMO.

Reply:
> If the user says it's 9 bytes, than 9 bytes it shall be IMO.

Yes, that's why I wrote the test.